core/src/main/java/org/apache/servicecomb/core/definition/OperationConfig.java

```diff
@@ -61,6 +61,20 @@
   @InjectProperty(keys = {"request.${op-any-priority}.timeout", "request.timeout"}, defaultValue = "30000")
   private long msRequestTimeout;
 
+  /**
+   * whether to remove certain headers from the 3rd party invocations
+   */
+  @InjectProperty(keys = {"request.${op-any-priority}.clientRequestHeaderFilterEnabled",
+      "request.clientRequestHeaderFilterEnabled"}, defaultValue = "true")
+  private boolean clientRequestHeaderFilterEnabled = true;
+
+  /**
+   * the matched headers should be removed, in invocations to 3rd party services
+   */
+  @InjectProperty(keys = {"request.${op-any-priority}.clientRequestHeaderFilterPattern",
+      "request.clientRequestHeaderFilterPattern"}, defaultValue = "^x-cse.*")
+  private String clientRequestHeaderFilterPattern = "^x-cse.*";
```

Comment:
if need to compile pattern, then we can do that in setter method
that can avoid check changes during main business logic.

Reply:
The logic is simplified. Now just remove all of the related headers, no regex pattern.